### PR TITLE
Expose AAD as an env var

### LINKED ISSUE #145: expose encryption identity of remote caller in secure context
currently, sender is exposed for the wallet signer of messages.
in a secure context, the key used for confidentiality should also be exposed to services, since it may live at a different life time from the sender wallet.

supercedes #48

### PR DESCRIPTION
**Pull request by abukosek**
Closes #145.

I'm not sure if I'm doing this right, so comments appreciated :)

TODO:
- [X] Fix tests.
- [X] Cleanup/rewrite, depending on how much I missed the target :D

### CODE REVIEW

File: ethcore/vm/src/action_params.rs [hunk 1]
```diff
@@ -96,6 +96,8 @@ pub struct ActionParams {
 	pub params_type: ParamsType,
 	/// The Oasis contract extracted from the code (if header is present).
 	pub oasis_contract: Option<OasisContract>,
+	/// The contract's session key used for encrypting/decrypting storage.
```
Comment (armaniferrante) [inline ethcore/vm/src/action_params.rs]: This comment isn't accurate. The session key doesn't belong to the contract. It belongs to the user initiating the transaction and defining the `data` payload in it.

File: ethcore/wasm/src/wasi.rs [hunk 2]
```diff
@@ -88,6 +88,18 @@ impl<'a> crate::Runtime<'a> {
 			format!("SENDER={:x}\0", self.context.sender).as_bytes(),
 		)?;
 		environs[2] = environ_buf;
+		let aad_env = match &self.context.aad {
+			Some(sk) => format!(
+				"AAD={}\0",
+				sk.iter()
+					.map(|byte| format!("{:02x}", byte))
```
Comment (abukosek) [inline ethcore/wasm/src/wasi.rs]: Going to change this to base64, as mentioned in the oasis-rs PR.

**Comment by peterjgilbert** [comment]
Thanks, @abukosek! This PR is currently exposing the peer's public key as the AAD env var. I assume we should switch this value to be the AAD obtained in `decrypt_session` (https://github.com/oasislabs/runtime-ethereum/blob/12072054f7569ce3d275f33bd1c5cad9510d790c/common/src/confidential/confidential_ctx.rs#L223). @nhynes / @armaniferrante, confirm?

**Comment by nhynes** [comment]
@peterjgilbert it has been done

related: https://github.com/oasislabs/runtime-ethereum/pull/851

**Review by nhynes — APPROVED** [review]

File: ethcore/vm/src/action_params.rs [hunk 3]
```diff
@@ -96,6 +96,8 @@ pub struct ActionParams {
 	pub params_type: ParamsType,
 	/// The Oasis contract extracted from the code (if header is present).
 	pub oasis_contract: Option<OasisContract>,
+	/// The AAD (Additional Authenticated Data) of confidential execution.
```
Comment (peterjgilbert) [inline ethcore/vm/src/action_params.rs]: Would be helpful to note that this is the AAD from the confidential transaction/request.

**Review by peterjgilbert — APPROVED** [review]

**Pull request merged**